pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -613,6 +619,55 @@ func (r *ReconcileComplianceRemediation) verifyAndCompleteMC(obj *unstructured.U
 	return nil
 }
 
+// Process kubeletconfig remediation
+func (r *ReconcileComplianceRemediation) verifyAndCompleteKC(obj *unstructured.Unstructured, rem *compv1alpha1.ComplianceRemediation) error {
+	scan := &compv1alpha1.ComplianceScan{}
+	scanKey := types.NamespacedName{Name: rem.Labels[compv1alpha1.ComplianceScanLabel], Namespace: rem.Namespace}
+	if err := r.client.Get(context.TODO(), scanKey, scan); err != nil {
+		return fmt.Errorf("couldn't get scan for KC remediation: %w", err)
+	}
+	mcfgpools := &mcfgv1.MachineConfigPoolList{}
+	if err := r.client.List(context.TODO(), mcfgpools); err != nil {
+		return fmt.Errorf("couldn't list the pools for the remediation: %w", err)
+	}
+	// The scans contain a nodeSelector that ultimately must match a machineConfigPool. The only way we can
+	// ensure it does is by checking if it matches any MachineConfigPool's labels.
+	// See also: https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md
+	ok, pool := utils.AnyMcfgPoolLabelMatches(scan.Spec.NodeSelector, mcfgpools)
+	if !ok {
+		return common.NewNonRetriableCtrlError("not applying remediation that doesn't have a matching MachineconfigPool. Scan: %s", scan.Name)
+	}
+
+	// We need to check if there is exsisting kubeletConfig present for selected MCP
+	hasCustomKC, kubletName, err := utils.IsMcfgPoolUsingKC(pool)
+
+	if err != nil {
+		//string to int convertion error
+		return fmt.Errorf("string-int convertion error for KC remediation: %w", err)
+	}
+	// we need to patch the remediation if there is already a CustomKC present
+	if hasCustomKC {
+		kubeletConfig := &mcfgv1.KubeletConfig{}
+		kcKey := types.NamespacedName{Name: kubletName, Namespace: rem.Namespace}
+		if err := r.client.Get(context.TODO(), kcKey, kubeletConfig); err != nil {
+			return fmt.Errorf("couldn't get current KubeletConfig: %w", err)
+		}
```

Comment:
Here you could set the name of the KC to patch:
```
        // Set kublet config name
	obj.SetName(kubeletConfig.GetName())
	obj.SetLabels(kubeletConfig.GetLabels())
```
And then just return. That way, when the object enters the reconcile loop, it would be handled appropriately.